label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -102,10 +103,17 @@ public void getLabelApplicationRules(
               .filter(rule -> !labelApplicationRuleIds.contains(rule.getId()))
               .filter(rule -> !deletedSystemLabelApplicationRuleIds.contains(rule.getId()))
               .collect(Collectors.toUnmodifiableList());
+      List<LabelApplicationRule> allLabelApplicationRules =
+          Stream.concat(
+                  labelApplicationRules.stream(), filteredSystemLabelApplicationRules.stream())
+              .filter(
+                  rule ->
+                      !request.hasFilter()
+                          || request.getFilter().getIdsList().contains(rule.getId()))
```

Comment:
prefer to just check if the id list is empty rather than if it has a filter. That way, if a second filter condition is added this logic still holds.